mesa/model.py

```diff
@@ -21,7 +22,15 @@ class Model:
     def __new__(cls, *args: Any, **kwargs: Any) -> Any:
         """Create a new model object and instantiate its RNG automatically."""
         obj = object.__new__(cls)
-        obj._seed = kwargs.get("seed", None)
+        obj._seed = kwargs.get("seed")
+        if obj._seed is None:
+            # We explicitly specify the seed here so that we know its value in
+            # advance. This value matches random.Random's default seed, which
+            # is the current timestamp. Note: datetime.now().timestamp() is
```

Comment:
It usually isn't the default seed.

> If a is omitted or None, the current system time is used. If randomness sources are provided by the operating system, they are used instead of the system time (see the [os.urandom()](https://docs.python.org/3/library/os.html#os.urandom) function for details on availability).

Its hard to find a OS without a randomness source.

Reply:
I have removed the statement that it the timestamp is CPython's default. This is how they do it: https://github.com/python/cpython/blob/d4cea794a7b9b745817d2bd982d35412aef04710/Modules/_randommodule.c#L283. There is a PyObject interface to it, but it is a non-public module, `_random.Random`.

Reply:
I would say simply random.random() would be the appropriate choice. That also wouldn't require any comment, because the code itself would already "say": just use a random value. And that's what we want. If someone requires cryptographically secure seeds they can provide them. But I can't really imagine a use case for that.